lang_js/analyze/js_to_generic.ml

```diff
@@ -121,16 +121,19 @@ let special (x, tok) =
           | [e] -> G.Await (tok, e)
           | _ -> error tok "Impossible: Too many arguments to Await"
           )
-  | Encaps v1 -> 
-      (match v1 with
-      | None -> SR_NeedArgs (fun args -> 
-          G.Call (G.IdSpecial (G.ConcatString G.InterpolatedConcat, tok), 
-                  args |> List.map (fun e -> G.Arg e)))
-      | Some n -> 
-            let n = name n in
-            SR_NeedArgs (fun args ->
-            G.OtherExpr (G.OE_EncapsName,(G.I n)::(args|>List.map(fun e ->G.E e))))
-      )
+  | Encaps has_tag_function -> 
+      if not has_tag_function 
+      then SR_NeedArgs (fun args -> 
+            G.Call (G.IdSpecial (G.ConcatString G.InterpolatedConcat, tok), 
+                    args |> List.map (fun e -> G.Arg e)))
```

Comment:
A nice-to-have here would be to throw the argument to `SR_NeedArgs` in a `with` and give it a nice name (`InterpolatedStringCall`?). Would also allow you to DRY lines 134-136.

Reply:
I don't understand this comment.